test/TestKeyValueSorter.cpp

```diff
@@ -0,0 +1,225 @@
+//////////////////////////////////////////////////////////////////////////////////////
+// Copyright 2020 Lawrence Livermore National Security, LLC and other CARE developers.
+// See the top-level LICENSE file for details.
+//
+// SPDX-License-Identifier: BSD-3-Clause
+//////////////////////////////////////////////////////////////////////////////////////
+
+#define GPU_ACTIVE
+
+#include "care/config.h"
+
+// other library headers
+#include "gtest/gtest.h"
+
+// care headers
+#include "care/array.h"
+#include "care/care.h"
+#include "care/KeyValueSorter.h"
+
+TEST(KeyValueSorter, SizeConstructor)
```

Comment:
Thanks for adding more tests

Reply:
This definitely made it easier to test what fixes were needed when implicit casts were enabled.